gateway/crates/gateway-binary/src/telemetry.rs

```diff
@@ -0,0 +1,102 @@
+use grafbase_telemetry::otel::opentelemetry_sdk::metrics::SdkMeterProvider;
+use tracing_subscriber::EnvFilter;
+
+use grafbase_telemetry::config::TelemetryConfig;
+use grafbase_telemetry::otel::layer::OtelTelemetry;
+use grafbase_telemetry::otel::opentelemetry_sdk::runtime::Tokio;
+use grafbase_telemetry::otel::opentelemetry_sdk::trace::TracerProvider;
+
+use crate::args::Args;
+
+#[derive(Default, Clone)]
+pub(crate) struct OpenTelemetryProviders {
+    pub meter: Option<SdkMeterProvider>,
+    pub tracer: Option<TracerProvider>,
+}
+
+pub(crate) fn init(args: &impl Args, config: Option<TelemetryConfig>) -> anyhow::Result<OpenTelemetryProviders> {
+    use tracing_subscriber::layer::SubscriberExt;
+    use tracing_subscriber::util::SubscriberInitExt;
+
+    let filter = args.log_level().map(|l| l.as_filter_str()).unwrap_or("info");
+    let env_filter = EnvFilter::new(filter);
+    let id_generator = {
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "lambda")] {
+                use opentelemetry_aws::trace::{XrayIdGenerator, XrayPropagator};
+                grafbase_telemetry::otel::opentelemetry::global::set_text_map_propagator(XrayPropagator::default());
+
+                XrayIdGenerator::default()
+            } else {
+                use grafbase_telemetry::otel::opentelemetry_sdk::trace::RandomIdGenerator;
+
+                RandomIdGenerator::default()
+            }
+        }
+    };
+
+    let config = config.unwrap_or(TelemetryConfig {
+        service_name: "unknown".to_string(),
+        resource_attributes: Default::default(),
+        tracing: Default::default(),
+        exporters: Default::default(),
+        logs: Default::default(),
+        metrics: Default::default(),
+        grafbase: Default::default(),
+    });
+
+    let OtelTelemetry {
+        tracer,
+        meter_provider,
+        logger,
+    } = grafbase_telemetry::otel::layer::build(config, id_generator, Tokio)?;
+
+    if let Some(ref meter_provider) = meter_provider {
+        grafbase_telemetry::otel::opentelemetry::global::set_meter_provider(meter_provider.clone());
+    }
+
+    if let Some(ref logger) = logger {
+        grafbase_telemetry::otel::opentelemetry::global::set_logger_provider(logger.provider.clone());
+    }
```

Comment:
I think this is not needed. It's removed in the latest otel crates, and worked fine just without when I tried.